core-backend/sandbox/src/funcs.rs

```diff
@@ -0,0 +1,472 @@
+use crate::env::Runtime;
+use alloc::{string::String, vec};
+use core::{
+    convert::{TryFrom, TryInto},
+    slice::Iter,
+};
+use gear_backend_common::{funcs, ExtInfo, WAIT_TRAP_STR};
+use gear_core::{
+    env::Ext,
+    message::{MessageId, OutgoingPacket, ReplyPacket},
+    program::ProgramId,
+};
+use sp_sandbox::{HostError, ReturnValue, Value};
+
+pub(crate) type SyscallOutput = Result<ReturnValue, HostError>;
+
+pub(crate) fn pop_i32<T: TryFrom<i32>>(arg: &mut Iter<'_, Value>) -> Result<T, HostError> {
```

Comment:
[nice](https://www.google.com/search?q=thank+you+gif+key+pelle&tbm=isch&ved=2ahUKEwju6tDPven1AhUMR8AKHWj7BfkQ2-cCegQIABAA&oq=thank+you+gif+key+pelle&gs_lcp=CgNpbWcQAzoKCAAQsQMQgwEQQzoFCAAQgAQ6BAgAEEM6BAgAEB46BAgAEBM6CAgAEAgQHhATOgYIABAIEB5QtwVY7xJgsBRoAHAAeACAAUOIAfoEkgECMTGYAQCgAQGqAQtnd3Mtd2l6LWltZ8ABAQ&sclient=img&ei=ruj-Ye7bLIyOgQbo9pfIDw&bih=968&biw=1920&client=ubuntu&hs=Duq#imgrc=HwqsTqL5ohayqM)